hmcts/pre-portal
# S28-2256: Users have to accept the Ts and Cs

> [!NOTE]
> This PR requires https://github.com/hmcts/pre-api/pull/701 and https://github.com/hmcts/pre-api/pull/696 to be merged before testing.

### JIRA ticket(s)

- S28-3258

### Change description

- Create new typed functions for better type safety
- Created page to display ts and cs in a box with an accept checkbox
- Added new function pre-cleint to send request to the API to accept the tc and cs for the current user
- Added redirect to the new ts and cs page for users who have not accepted the tc and cs
- Grab the terms content from the api



> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**

Note: [pre-api#497](https://github.com/hmcts/pre-api/pull/497) needs testing and merging before this PR.

- Create a new account
- The when trying to access `/browse` user should be redirected to `/accept-terms-and-conditions`
- Accept the ts ans cs
- User should then be redirected to `/browse`
